packages/ember-data/lib/system/store.js

```diff
@@ -1683,6 +1683,10 @@ Store = Service.extend({
       return internalModels.map((internalModel) => internalModel.getRecord());
     }
 
+    if (data.data === null) {
+      return null;
+    }
+
     var internalModel = this._pushInternalModel(data.data || data);
```

Comment:
I think this special case (I've never really liked it) should be considered if we return early if `data.data === null` above.